core/templates/domain/exploration/editable-exploration-backend-api.service.spec.ts

```diff
@@ -95,180 +73,182 @@ describe('Editable exploration backend API service', function() {
+      expect(req.request.method).toEqual('GET');
+      req.flush(sampleDataResults);
+      flushMicrotasks();
 
       expect(successHandler).toHaveBeenCalledWith(sampleDataResults);
       expect(failHandler).not.toHaveBeenCalled();
-    }
+    })
   );
 
   it('should fetch and apply the draft of an exploration',
-    function() {
+    fakeAsync(() => {
       var successHandler = jasmine.createSpy('success');
       var failHandler = jasmine.createSpy('fail');
-
       // Loading a exploration the first time should fetch it from the backend.
-      $httpBackend.expect(
-        'GET', '/createhandler/data/0?apply_draft=true').respond(
-        sampleDataResults);
-
-      EditableExplorationBackendApiService.fetchApplyDraftExploration(
+      editableExplorationBackendApiService.fetchApplyDraftExploration(
         '0').then(successHandler, failHandler);
-      $httpBackend.flush();
+      let req = httpTestingController.expectOne(
+        '/createhandler/data/0?apply_draft=true');
+      expect(req.request.method).toEqual('GET');
+      req.flush(sampleDataResults);
+      flushMicrotasks();
 
       expect(successHandler).toHaveBeenCalledWith(sampleDataResults);
       expect(failHandler).not.toHaveBeenCalled();
-    }
+    })
   );
 
   it('should use the rejection handler if the backend request failed',
-    function() {
+    fakeAsync(() => {
       var successHandler = jasmine.createSpy('success');
       var failHandler = jasmine.createSpy('fail');
 
       // Loading a exploration the first time should fetch it from the backend.
-      $httpBackend.expect('GET', '/createhandler/data/1').respond(
-        500, 'Error loading exploration 1.');
-      EditableExplorationBackendApiService.fetchExploration('1').then(
+      editableExplorationBackendApiService.fetchExploration('1').then(
         successHandler, failHandler);
-      $httpBackend.flush();
+      let req = httpTestingController.expectOne('/createhandler/data/1');
+      expect(req.request.method).toEqual('GET');
+      req.flush('Error loading exploration 1.', {
+        status: 500, statusText: 'Invalid request'
+      });
+      flushMicrotasks();
 
       expect(successHandler).not.toHaveBeenCalled();
       expect(failHandler).toHaveBeenCalledWith('Error loading exploration 1.');
-    }
+    })
   );
 
   it('should update a exploration after fetching it from the backend',
-    function() {
+    fakeAsync(() => {
       var successHandler = jasmine.createSpy('success');
       var failHandler = jasmine.createSpy('fail');
       var exploration = null;
 
       // Loading a exploration the first time should fetch it from the backend.
-      $httpBackend.expect('GET', '/createhandler/data/0').respond(
-        sampleDataResults);
-
-      EditableExplorationBackendApiService.fetchExploration('0').then(
+      editableExplorationBackendApiService.fetchExploration('0').then(
         function(data) {
```

Comment:
Can we use arrow functions here and in other places?

Reply:
Done